mesonbuild/compilers/rust.py

```diff
@@ -301,6 +302,14 @@ def get_rust_tool(self, name: str, env: Environment) -> T.List[str]:
 
         return exelist + args
 
+    def has_multi_arguments(self, args: T.List[str], env: 'Environment') -> T.Tuple[bool, bool]:
+        return self.compiles('fn main { std::process::exit(0) };\n', env, extra_args=args, mode=CompileCheckMode.COMPILE)
+
+    def has_multi_link_arguments(self, args: T.List[str], env: 'Environment') -> T.Tuple[bool, bool]:
+        args = self.linker.fatal_warnings() + args
```

Comment:
From opening up the PR posted by @pks-t, one thing I immediately noticed right at the start is: all these quoted annotations! :P

Reply:
Nah, I looked at the first iteration of this PR and when it was originally opened this file had `from __future__ import annotations`.

Reply:
I meant more, it is so old we still had a ton of quoted stuff to copy past from.